libclc/libspirv/lib/native_cpu/workitem/get_global_id.cl

```diff
@@ -10,14 +10,15 @@
 
 ulong __mux_get_global_id(int);
 
-_CLC_DEF _CLC_OVERLOAD size_t __spirv_GlobalInvocationId_x() {
-  return __mux_get_global_id(0);
-}
-
-_CLC_DEF _CLC_OVERLOAD size_t __spirv_GlobalInvocationId_y() {
-  return __mux_get_global_id(1);
-}
-
-_CLC_DEF _CLC_OVERLOAD size_t __spirv_GlobalInvocationId_z() {
-  return __mux_get_global_id(2);
+_CLC_DEF _CLC_OVERLOAD size_t __spirv_BuiltInGlobalInvocationId(int dim) {
+  switch (dim) {
+  case 0:
+    return __mux_get_global_id(0);
+  case 1:
+    return __mux_get_global_id(1);
+  case 2:
+    return __mux_get_global_id(2);
+  default:
+    return 0;
+  }
```

Comment:
Can we just do
```suggestion
  return __mux_get_global_id(dim);
```
? (This looks like this comment also applies to several other functions that are not for Native CPU.)

Reply:
> ? (This looks like this comment also applies to several other functions that are not for Native CPU.)

done, thanks. I kept switch-case for a few functions where data is read from buffer.